scripts/hud/jump-stats.ts

```diff
@@ -1,12 +1,41 @@
-class JumpStats {
-	/** @type {Label} @static */
-	static label = $('#JumpStatsLabel');
-	/** @type {Panel} @static */
-	static container = $('#JumpStatsContainer');
-	/** @type {Panel} @static */
-	static panel = $.GetContextPanel();
-
-	static onJump() {
+import { PanelHandler } from 'util/module-helpers';
+import { RegisterHUDPanelForGamemode } from '../util/register-for-gamemodes';
+import { GamemodeCategories, GamemodeCategory } from '../common/web';
+
+@PanelHandler()
+class JumpStatsHandler {
+	readonly panels = {
+		cp: $.GetContextPanel<MomHudJumpStats>(),
+		label: $<Label>('#JumpStatsLabel'),
+		container: $<MomHudJumpStats>('#JumpStatsContainer')
+	};
+
+	jumpStatsConfig: typeof this.panels.cp.jumpStatsCFG;
+
+	bufferLength: number;
+	countBuffer: string[];
+	takeoffSpeedBuffer: string[];
+	speedDeltaBuffer: string[];
+	takeoffTimeBuffer: string[];
+	timeDeltaBuffer: string[];
+	strafesBuffer: string[];
+	syncBuffer: string[];
+	gainBuffer: string[];
+	yawRatioBuffer: string[];
+	heightDeltaBuffer: string[];
+	distanceBuffer: string[];
+	efficiencyBuffer: string[];
+
+	constructor() {
+		RegisterHUDPanelForGamemode({
+			gamemodes: GamemodeCategories.get(GamemodeCategory.BHOP),
+			onLoad: () => this.onMapInit(),
+			handledEvents: [{ event: 'OnJumpStarted', panel: this.panels.container, callback: () => this.onJump() }]
+		});
+		$.RegisterForUnhandledEvent('OnJumpStatsCFGChange', () => this.onConfigChange());
```

Comment:
should this be listed in `events` when registering the panel for bhop?

Reply:
Don't think so, as presumably we want to handle changes even if they happen when playing other modes. Or does the entire config get loaded from scratch when loading into a map?

Reply:
I think in every case where we would be registering a panel for a specific mode, the config settings get loaded as a part of `OnMapInit()`

Reply:
Good to know, I'll move everything in this case and leave a comment clarifying